[ADVAPP-568]: Add light documentation on new permission creation (#811)
canyongbs/advisingapp: docs/roles-and-permissions.md

This application uses the [Spatie Laravel-permission](https://spatie.be/docs/laravel-permission/v5/introduction) package in order to facilitate the usage of roles and permissions.

The application uses a dedicated convention in order to define and populate roles and permissions, leaving little in the way of thinking when it comes time to stand up the application locally or in a live environment.

### Local Setup
In order to get your local environment correctly set up, you won't have to do anything beyond running the setup process defined in the documentation. Permissions are created in data migrations and roles are created by seeders when a Tenant is created.

## Permissions

> Permissions in the application are created and managed through special data Migrations called Permission Migrations. Details on Data Migrations can be found in the [Data Migrations](/docs/data-migrations.md) documentation.

Reviewer: sorry

```suggestion
Permissions in the application are created and managed through special Data Migrations called Permission Migrations. Details on Data Migrations can be found in the [Data Migrations](/docs/data-migrations.md) documentation.
```

### Permission Migrations
Permission Migrations operate much the same as regular Data Migrations but adhere to a few slightly different or additional rules:

1. Prepend the name of the migration with `seed_permissions_` to differentiate it from schema migrations and Data Migrations.

> 2. The `Database\Migrations\Concerns\CanModifyPermissions` Trait and it's available helpers should be used.

Reviewer: ```suggestion
2. Use the helpers in the `Database\Migrations\Concerns\CanModifyPermissions` trait.
```

> 3. Any queries should be surrounded by both a database transaction and a `try/catch` block to catch any SQL errors such as a `UniqueConstraintViolation`. This is to prevent issues with permission duplication and the corruption of the current connection if such an error is triggered.

Reviewer: ```suggestion
3. Any other queries you write should be surrounded by both a database transaction and a `try`/`catch` block to catch any SQL errors such as a `UniqueConstraintViolation`.
```

Removing that second bit because its now handled by the `insertOrIgnore()`
